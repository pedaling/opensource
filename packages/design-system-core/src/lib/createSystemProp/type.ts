import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
