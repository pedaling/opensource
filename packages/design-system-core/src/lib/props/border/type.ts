import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
