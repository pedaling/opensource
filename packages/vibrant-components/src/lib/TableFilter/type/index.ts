export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
