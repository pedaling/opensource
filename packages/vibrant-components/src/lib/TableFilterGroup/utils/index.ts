import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
