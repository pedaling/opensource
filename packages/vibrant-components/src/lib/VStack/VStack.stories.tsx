import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Box } from '@vibrant-ui/core';
import { VStack } from './VStack';

export default {
  title: 'VStack',
  component: VStack,
  args: {
    spacing: 20,
  },
} as ComponentMeta<typeof VStack>;

export const Basic: ComponentStory<typeof VStack> = props => (
  <VStack {...props}>
    <Box width={200} height={200} backgroundColor="primary" />
    <Box width={200} height={200} backgroundColor="primary" />
  </VStack>
);
