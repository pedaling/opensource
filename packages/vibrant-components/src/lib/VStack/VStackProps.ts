import { withVariation } from '@vibrant-ui/core';
import type { StackProps } from '../Stack';

export type VStackProps = Omit<StackProps, 'direction'> & {
  alignHorizontal?: StackProps['alignItems'];
  alignVertical?: StackProps['justifyContent'];
};

export const withVStackVariation = withVariation<VStackProps>('VStack')();
