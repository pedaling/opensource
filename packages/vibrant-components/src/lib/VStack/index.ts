export { VStack } from './VStack';
export type { VStackProps } from './VStackProps';
