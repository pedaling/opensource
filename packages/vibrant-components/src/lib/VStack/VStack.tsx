import { Stack } from '../Stack';
import { withVStackVariation } from './VStackProps';

export const VStack = withVStackVariation(({ innerRef, alignHorizontal, alignVertical, ...stackProps }) => (
  <Stack
    direction="vertical"
    ref={innerRef}
    {...stackProps}
    justifyContent={alignVertical}
    alignItems={alignHorizontal}
  />
));
