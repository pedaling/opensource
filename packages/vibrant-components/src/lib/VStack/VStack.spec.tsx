import { Box } from '@vibrant-ui/core';
import type { ReactRenderer } from '@vibrant-ui/utils/testing';
import { createReactRenderer } from '@vibrant-ui/utils/testing';
import { VStack } from './VStack';

describe('<VStack />', () => {
  const { render } = createReactRenderer();
  let renderer: ReactRenderer;
  let element: HTMLElement;

  describe('when VStack rendered', () => {
    beforeEach(async () => {
      renderer = render(
        <VStack data-testid="vstack" alignVertical="center">
          <Box width={20} height={20} />
          <Box width={20} height={20} />
        </VStack>
      );

      element = await renderer.findByTestId('vstack');
    });

    it('stack alignItem is center', () => {
      expect(element).toHaveStyleRule('justify-content', 'center');
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when VStack vertical and horizontal alignment varies', () => {
    beforeEach(async () => {
      renderer = render(
        <VStack data-testid="hstack-alignment" alignVertical="space-between" alignHorizontal="start">
          <Box width={20} height={20} />
          <Box width={20} height={20} />
        </VStack>
      );

      element = await renderer.findByTestId('hstack-alignment');
    });

    it('stack align-items is start', () => {
      expect(element).toHaveStyleRule('align-items', 'flex-start');
    });

    it('stack justify-content is space-between', () => {
      expect(element).toHaveStyleRule('justify-content', 'space-between');
    });
  });
});
