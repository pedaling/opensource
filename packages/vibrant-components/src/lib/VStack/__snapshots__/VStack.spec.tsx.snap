// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VStack /> when VStack rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  box-sizing: border-box;
  position: relative;
  width: 20px;
  height: 20px;
}

<div>
  <div
    class="emotion-0"
    data-testid="vstack"
  >
    <div
      class="emotion-1"
    />
    <div
      class="emotion-1"
    />
  </div>
</div>
`;
